### ISSUE TITLE
Version information in the browser_protocol.json and `js_protocol.json` are not useful.

### ISSUE THREAD

**Issue by fake-name** [ChromeDevTools/devtools-protocol#6]
> Moved from https://bugs.chromium.org/p/chromium/issues/detail?id=706677#

Basically, [I'm writing some tooling](https://github.com/fake-name/ChromeController) that interacts with the chromium remote debugger interface. The remote debugger interface has a json file that specifies it's methods, and also has a version number in it. That version number hasn't been changed while features have been added and removed repeatedly.

> Aside: I'm basically trying to implement basically a complete toolset for remotely controlling the browser (e.g. something like PhantomJS without selenium/webdriver and all the pain and [spectacularly asinine missing features](http://stackoverflow.com/questions/14537336/checking-httpresponse-ok-200-with-selenium-webdriver) that brings). As such, I find myself needing to touch a bunch of the functionality marked "experimental" in the protocol file.

I'd expect changes to the remote debugging interface to be associated with a change of the remote debugging interface number. This would be a clear indicator that I need to update my interface, and would also provide a definitive way to refer to one version of the interface specification when dealing with bugs in either my software or chromium.

As it is, I have no idea why the version number is even there, since it seems like it doesn't ever get changed or updated.

It'd probably be nice to make it clear what the version information means, or better yet, allow it to be used for meaningful versioning. I'd love if there was a further sub-version-number that was incremented for *any* change to the interface structure.

**Comment by paulirish** [comment]
TL;DR: the "1.2" version in https://github.com/ChromeDevTools/devtools-protocol/blob/master/json/browser_protocol.json#L2 doesn't follow any particular versioning philosophy so it's difficult to write tooling against.

**Comment by fake-name** [comment]
Any movement on this?

**Comment by paulirish** [comment]
the versions from the commits are chromium revision numbers. 

the `Browser.getVersion` method returns data including a chromium commit hash. https://omahaproxy.appspot.com/ has APIs that can turn that hash into a chromium revision. (I think it'll take two API calls..)

However you probably dont need to do all that. `/json/protocol` now serves the protocol that your chrome speaks. This works as of Chrome 60. https://codereview.chromium.org/2867073006/

**Comment by fake-name** [comment]
Oh, cool, that happened? Awesome!

IIRC, there was discussion of having chrome serve a version of the protocol, but it died because it increased the chrome binary size.

------

As an aside, there *still* seems to be no way to get the browser version via the debug interface. This has been quite problematic, as even filing bug reports generally requires some sort of version. I've been using git commit hashes, but that's just lead to further confusion.

**Comment by paulirish** [comment]
you can get chrome version and git hash. chromium revision requires those API calls, but that probably won't be changing soon.

**Comment by fake-name** [comment]
Derp, apparently I need to learn to read the literal comment I'm replying to.

**Comment by fake-name** [comment]
Ok, so there's a new version (1.3), and the version that tracks HEAD.

However, the version tag in the file has been completely removed. What?

**Comment by TimvdLippe** [comment]
This repository is related to Chrome DevTools Protocol, but does not track issues regarding its definition or implementation. If you want to file an issue for the Chrome DevTools Protocol, please open an issue on https://crbug.com under [`component: Platform>DevTools>Platform`](https://bugs.chromium.org/p/chromium/issues/entry?components=Platform%3EDevTools%3EPlatform). Thanks in advance!

**Comment by fake-name** [comment]
.... What's the point of this repo, then?

**Comment by TimvdLippe** [comment]
Please see https://github.com/ChromeDevTools/devtools-protocol/pull/209 (and the existing README)

**Comment by fake-name** [comment]
If you're not intending to accept bug reports here, you can turn the issue tracker in this repo off, you know.

**Comment by TimvdLippe** [comment]
> This repository only hosts issues regarding the type definitions of CDP.
